projects/js-toolkit/docs/manuals/adaptation-process.md

```diff
@@ -0,0 +1,208 @@
+
+This is necessary since the JS Toolkit needs to know what to attach to the
+portlet's DOM node so that your Angular application is rendered inside it.
+
+So, your application's component should look like this:
+
+```javascript
+@Component({
+  selector: 'app-root',
+  …
+})
+export class AppComponent {
+  …
+}
+```
+
+When `build:liferay` is run, the standard webpack based build of `Angular CLI`
+is invoked, which leaves processed `.js` and `.css` files inside the `dist`
+directory.
+
+These files are then tweaked by the adapter scripts so that they can be
+launched from Liferay's standard entry point (see
+[JavaScript portlet entry point](../reference/js-portlet-entry-point.md)).
+
+To achieve that, the `app-root` selector in the output bundles will be changed
+so that it points to the portlet's main `<div>` node (identified by the
+`portletElementId` parameter of the Liferay's entry point).
+
+All this happens _automagically_ under the hood when the `build:liferay` script
+is run so that you don't need to do anything other than adapting to the
+expected code structure.
+
+### Vue CLI projects
+
+#### Detection
+
+Any project containing `@vue/cli-service` as a dependency or devDependency is
+recognized as a `Vue CLI` project.
+
+#### Expected structure
+
+These projects are expected to follow the standard `Vue CLI` structure but, in
+addition, they must use `#app` as their application's DOM selector.
+
+This is necessary since the JS Toolkit needs to know what to attach to the
+portlet's DOM node so that your Vue application is rendered inside it.
+
+So, your application's mount code should look like this:
+
+```javascript
+new Vue({
+	render: (h) => h(App),
+}).$mount('#app');
+```
+
+When `build:liferay` is run, the standard webpack based build of `Vue CLI` is
+invoked, which leaves processed `.js` and `.css` files inside the `dist`
+directory.
+
+These files are then tweaked by the adapter scripts so that they can be
+launched from Liferay's standard entry point (see
+[JavaScript portlet entry point](../reference/js-portlet-entry-point.md)).
+
+To achieve that, the `#app` selector in the output bundles will be changed so
+that it points to the portlet's main `<div>` node (identified by the
+`portletElementId` parameter of the Liferay's entry point).
+
+All this happens _automagically_ under the hood when the `build:liferay` script
+is run so that you don't need to do anything other than adapting to the
+expected code structure.
+
+## Limitations
+
+Given that adaptation is a somewhat heuristic process in which the bundler does
+its best effort to make everything work like in the project framework's native
+build, there are (and will be) some [limitations](../caveats/adaptation.md).
+
+## One final note
+
+The general idea of the adaptation is to be able to deploy _native_ framework
+projects to Liferay. However, because you are deploying to Liferay, you may be
+able to use some of its APIs or features. But keep in mind that, the farther
```

Comment:
Should this be "may **not**"?

Reply:
I think it is not "not", but the "However", that is incorrect.

So, going to commit something like this:

```
The general idea of the adaptation is to be able to deploy _native_ framework
projects to Liferay. Because you are deploying to Liferay, you may be able to
use some of its APIs or features, but keep in mind that, the farther you drift
from the framework's structure, the more tied you will be to Liferay's platform
so it is worth considering migrating instead of adapting if you just want to
deploy to Liferay platform.
```